crates/benda/src/parser/mod.rs

```diff
@@ -0,0 +1,958 @@
+#![allow(clippy::cmp_owned)]
+use core::panic;
+use std::vec;
+
+use bend::fun::{self, Adt, Book, CtrField, Name, Op, Rule, STRINGS};
+use bend::imp::{self, Expr, MatchArm, Stmt};
+use indexmap::IndexMap;
+use num_traits::cast::ToPrimitive;
+use rustpython_parser::ast::{
+    located, CmpOp as rCmpOp, Expr as rExpr, ExprAttribute, ExprBinOp,
+    Operator as rOperator, Pattern as rPattern, Stmt as rStmt, StmtAssign,
+    StmtClassDef, StmtExpr, StmtFunctionDef, StmtIf, StmtMatch,
+};
+
+use crate::benda_ffi::run;
+
+#[derive(Clone, Debug)]
+enum FromExpr {
+    CtrField(Vec<CtrField>),
+    Expr(imp::Expr),
+    Statement(imp::Stmt),
+}
+
+impl FromExpr {
+    pub fn to_expr(&self) -> Option<Expr> {
+        if let FromExpr::Expr(exp) = self {
+            return Some(exp.clone());
+        }
+        None
+    }
+
+    pub fn to_stmt(&self) -> Option<Stmt> {
+        if let FromExpr::Statement(stmt) = self {
+            return Some(stmt.clone());
+        }
+        None
+    }
+
+    pub fn get_var_name(&self) -> Option<Name> {
+        if let FromExpr::Expr(Expr::Var { nam }) = self {
+            Some(nam.clone())
+        } else {
+            None
+        }
+    }
+}
```

Comment:
Looks redundant 🤔

Reply:
I tried to remove the excessive use of pattern matching in places in the code where I knew the expected type. I'll find a better way to represent this.